contracts/helpers/MultiBlockHarvester.sol

```diff
@@ -0,0 +1,213 @@
+        _adjustAllowance(yieldBearingAsset, address(transmuter), balance);
+        uint256 amountOut = transmuter.swapExactInput(
+            balance,
+            0,
+            yieldBearingAsset,
+            address(agToken),
+            address(this),
+            block.timestamp
+        );
+        address depositAddress = yieldBearingAsset == XEVT
+            ? yieldBearingToDepositAddress[yieldBearingAsset]
+            : address(0);
+        _checkSlippage(balance, amountOut, yieldBearingAsset, depositAddress, true);
+        agToken.burnSelf(amountOut, address(this));
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+                                                        INTERNAL FUNCTIONS
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
+
+    function _rebalance(
+        uint8 typeAction,
+        address yieldBearingAsset,
+        YieldBearingParams memory yieldBearingInfo,
+        uint256 amount
+    ) internal {
+        if (amount > maxMintAmount) revert TooBigAmountIn();
+        agToken.mint(address(this), amount);
+        _adjustAllowance(address(agToken), address(transmuter), amount);
+        if (typeAction == 1) {
+            address depositAddress = yieldBearingToDepositAddress[yieldBearingAsset];
+
+            if (yieldBearingAsset == XEVT) {
+                uint256 amountOut = transmuter.swapExactInput(
+                    amount,
+                    0,
+                    address(agToken),
+                    yieldBearingInfo.stablecoin,
+                    address(this),
+                    block.timestamp
+                );
+                _checkSlippage(amount, amountOut, yieldBearingInfo.stablecoin, depositAddress, false);
+                _adjustAllowance(yieldBearingInfo.stablecoin, address(depositAddress), amountOut);
+                (uint256 shares, ) = IPool(depositAddress).deposit(amountOut, address(this));
+                _adjustAllowance(yieldBearingAsset, address(transmuter), shares);
+                amountOut = transmuter.swapExactInput(
+                    shares,
+                    0,
+                    yieldBearingAsset,
+                    address(agToken),
+                    address(this),
+                    block.timestamp
+                );
+                agToken.burnSelf(amountOut, address(this));
+            } else if (yieldBearingAsset == USDM) {
+                uint256 amountOut = transmuter.swapExactInput(
+                    amount,
+                    0,
+                    address(agToken),
+                    yieldBearingInfo.stablecoin,
+                    address(this),
+                    block.timestamp
+                );
+                _checkSlippage(amount, amountOut, yieldBearingInfo.stablecoin, depositAddress, false);
```

Comment:
Seems there is some redundancy in this part, the `amountOut` calculation and `_checkSlippage` are repeated in both if condition. For a better readability, maybe consider a little refactor and put this part above the if.